### PR TITLE
update CXCompatible

### PR DESCRIPTION
**Pull request by ddddxxx**
Current Combine compatible design is inherently broken:

1. We have different kinds of wrapper type: `AnyCXWrapper<T>`, `AnyObjectCXWrapper<T>`, `DispatchQueueCXWrapper` etc. Their name is unpredictable and confusing.
2. Type system is a mass. We have `Optional.CX`, `NSObject.CX`, `DispatchQueue.CX`, but not `JSONDecoder.CX`. Type of `dispatchQueue.cx` is not `DispatchQueue.CX`, instead, it's a wrapper type.
3. We have to introduce `ResultProtocol` to support `Result.Publisher`.

I propose a new Combine compatible design with a small set of simple and consistent rule:

1. `CXWrapper` must be a concrete type. Generic wrapper type is not allowed.
2. Type of `object.cx` must be `Object.CX`.
3. Introduce namespace `CXWrappers`. Wrapper type of a `Type` must be `Type.CX` and `CXWrappers.Type`, They must be aliases to each other. (in case of ambiguity between NSObject and its subclasses)
4. Deprecate static `.cx` property that reference to meta type of the wrapper. Since `Type.CX` is always available.
5. For CXCompatible, `.CX` and `.cx` simply return itself. No wrapper type.

As a result, `OptionalProtocol` and `ResultProtocol` is removed from public interface.

The syntax is almost identical. Except `Timer.cx.publish(...)` becomes to `Timer.CX.publish(...)`.

### CODE REVIEW

**Comment by ddddxxx** [comment]
Wow I can't believe it passed the ci for the first time! considering the massive change I made.

File: Sources/CXCompatible/Foundation.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+extension CXWrappers {
+    typealias NSObject = Foundation.NSObject
+    typealias JSONEncoder = Foundation.JSONEncoder
+    typealias JSONDecoder = Foundation.JSONDecoder
+    typealias NotificationCenter = Foundation.NotificationCenter
+    typealias OperationQueue = Foundation.OperationQueue
+    typealias RunLoop = Foundation.RunLoop
+    typealias Timer = Foundation.Timer
+    typealias URLSession = Foundation.URLSession
```
Comment (luoxiu) [inline Sources/CXCompatible/Foundation.swift]: 这些不需要 public 吗？

**Review by luoxiu — COMMENTED** [review]
It looks good overall! Thanks again! @ddddxxx 

It seems that some types need `public`, don't they?

**Comment by ddddxxx** [comment]
Done!

**Pull request merged**